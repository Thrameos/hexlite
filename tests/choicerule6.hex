x.
y :- x.

d(1). d(2). d(3).

% variable disjunctive heads have different semantics than choice rules

% variable disjunctive head: rule provides only support for one atom
p(X) : d(X) :- x, y.
p(1) :- p(2).	% also check if head cycles work
p(2) :- p(1).

% choice rule: each atom can be individually true or false
{ q(X) : d(X) } :- x, y.
