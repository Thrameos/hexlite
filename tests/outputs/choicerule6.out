{x,y,d(1),d(2),d(3),p(1),p(2),q(1),q(2),q(3)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(1),q(2)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(1),q(3)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(1)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(2),q(3)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(2)}
{x,y,d(1),d(2),d(3),p(1),p(2),q(3)}
{x,y,d(1),d(2),d(3),p(1),p(2)}
{x,y,d(1),d(2),d(3),q(1),q(2),q(3),p(3)}
{x,y,d(1),d(2),d(3),q(1),q(2),p(3)}
{x,y,d(1),d(2),d(3),q(1),q(3),p(3)}
{x,y,d(1),d(2),d(3),q(1),p(3)}
{x,y,d(1),d(2),d(3),q(2),q(3),p(3)}
{x,y,d(1),d(2),d(3),q(2),p(3)}
{x,y,d(1),d(2),d(3),q(3),p(3)}
{x,y,d(1),d(2),d(3),p(3)}
